pkg/network/render.go

```diff
@@ -385,6 +388,61 @@ func isNetworkChangeSafe(prev, next *operv1.NetworkSpec, infraRes *bootstrap.Inf
 	return nil
 }
 
+func isClusterNetworkChangeSafe(prev, next *operv1.NetworkSpec) error {
+
+	// quick check to make sure clusterNetwork slices are of same size as we do not
+	// support adding/removing additional clusterNetwork entries unless it's for a
+	// single/dual stack migration. in those cases validation is done in isNetworkChangeSafe()
+	if len(prev.ClusterNetwork) != len(next.ClusterNetwork) {
+		return errors.Errorf("Adding/Removing clusterNetwork entries of the same type is not supported")
+	}
+
+	// Only support changing ClusterNetwork CIDR if it's OVNK
+	if !reflect.DeepEqual(next.DefaultNetwork.Type, operv1.NetworkTypeOVNKubernetes) {
+		return errors.Errorf("Adding/Removing clusterNetwork entries of the same type is not supported")
+	}
+
+	// sort prev and next just in case there was some re-ordering of the slice, since we
+	// want to know for sure we are comparing the same elements in each below
+	sort.SliceStable(prev.ClusterNetwork, func(i, j int) bool {
+		return prev.ClusterNetwork[i].CIDR < prev.ClusterNetwork[j].CIDR
+	})
+	sort.SliceStable(next.ClusterNetwork, func(i, j int) bool {
+		return next.ClusterNetwork[i].CIDR < next.ClusterNetwork[j].CIDR
+	})
+
+	// since we do not allow the clusterNetwork[] size to change, it should be safe to compare
+	// prev[i] to next[i] in this validation
+	for i, e := range prev.ClusterNetwork {
+		prevIp, prevMask, err := net.ParseCIDR(e.CIDR)
+		if err != nil {
+			return errors.Errorf("Error parsing CIDR from ClusterNetwork entry %s", e.CIDR)
```

Comment:
it would be good to wrap the error returned from `net.ParseCIDR` into the returned one.